pkg/models/error.go

```diff
@@ -158,3 +184,30 @@ func (e *BaseError) Details() map[string]string {
 func (e *BaseError) Code() ErrorCode {
 	return e.code
 }
+
+// HTTPStatusCode is a method that returns the httpStatusCode field of BaseError.
+// If no specific HTTP status code has been set, it returns 0.
+// This method can be used to retrieve the HTTP status code associated with the error,
+// which is useful when translating the error into an HTTP response.
+func (e *BaseError) HTTPStatusCode() int {
+	return e.httpStatusCode
+}
+
+func InferHTTPStatusCode(code ErrorCode) int {
+	switch code {
```

Comment:
this should be private and callers should only use `HTTPStatusCode()`

Reply:
Done